Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2007-12-28
Initial Package Version: 0.16.1
Upstream Status: Partially From Upstream
Origin: http://uclibc.org/cgi-bin/viewcvs.cgi/*checkout*/trunk/buildroot/package/gettext/gettext-error_print_progname.patch?rev=18866
http://cvs.savannah.gnu.org/viewvc/gnulib/lib/error.h?root=gnulib&view=markup
Description: This patch fixes build problems with uClibc. uClibc does not
feature the error_print_progname() library function, so we make a dummy
function.

Also see: http://linuxfromscratch.org/pipermail/hlfs-dev/2007-July/003676.html

--- gettext-0.16.1.orig/gettext-tools/gnulib-lib/error.h	2006-11-27 18:14:50.000000000 +0100
+++ gettext-0.16.1/gettext-tools/gnulib-lib/error.h	2007-06-20 13:29:32.000000000 +0200
@@ -50,7 +50,10 @@ extern void error_at_line (int __status,
 /* If NULL, error will flush stdout, then print on stderr the program
    name, a colon and a space.  Otherwise, error will call this
    function without parameters instead.  */
-extern DLL_VARIABLE void (*error_print_progname) (void);
+#ifndef __UCLIBC__
+extern DLL_VARIABLE
+#endif
+void (*error_print_progname) (void);
 
 /* This variable is incremented each time `error' is called.  */
 extern DLL_VARIABLE unsigned int error_message_count;
